packages/server/src/process_move_service.ts

```diff
@@ -0,0 +1,97 @@
+import { Board, Marker, Winner } from "@mapistry/take-home-challenge-shared";
+import { WinningLines } from "./constants";
+
+export class ProcessMoveService {
+    board: Board;
+
+    constructor(board: Board) {
+        this.board = board;
+    }
+
+    static calculateMinimaxScore(board: Board, depth: number, aiTurn: boolean) {
+        const scoreMap = {
+            O: 10,
+            X: -10,
+            tie: 0
+        }
+
+        let finalScore!: number;
```

Comment:
I added the non-null assertion here because a score is always returned from this recursive function call, but the compiler was having difficulty believing me. I'd love feedback on this approach.